src/NServiceBus.Metrics.PerformanceCounters.Tests/PerformanceCountersCacheTests.cs

```diff
@@ -0,0 +1,64 @@
+﻿namespace Tests
+{
+    using System;
+    using ApprovalTests;
+    using NUnit.Framework;
+
+    [TestFixture]
+    public class PerformanceCountersCacheTests
+    {
+        [Test]
+        public void Get_the_same_counter_twice_returns_the_same()
+        {
+            var cache = new TestablePerformanceCountersCache();
+
+            var counterName = "CounterName";
+            var instanceName = "InstancenName";
+
+            var firstCounter = cache.Get(new CounterInstanceName(counterName, instanceName));
+            var secondCounter = cache.Get(new CounterInstanceName(counterName, instanceName));
+
+            Assert.NotNull(firstCounter);
```

Comment:
`Assert.AreSame` calls into `Is.Same` which uses the `SameConstraint`. `SameConstraint` uses [ReferenceEquals](https://referencesource.microsoft.com/#mscorlib/system/object.cs,82). `null == null` is `true`